src/modelmatrix.jl

```diff
@@ -65,7 +65,7 @@ function modelmat_cols(::Type{T},
     ## are the same by constructing a re-indexing vector. Indexing into
     ## reindex with v.refs will give the corresponding row number of the
     ## contrast matrix
-    reindex = [findfirst(x -> x == contrast.levels, l) for l in levels(v)]
+    reindex = [findfirst(x -> x == l, contrast.levels) for l in levels(v)]
```

Comment:
For an appropriate Compat version, this can be `findfirst(equalto(l), contrast.levels)`

Reply:
Yeah, why not, though the current version isn't terrible either.

Reply:
The only difference is that `equalto` uses `isequal`. Probably shouldn't matter in this case, unless any of the levels are `NaN` or `missing` or some other irregular thing.